### PR TITLE
Exercise: Mother Child Age Difference

### PR DESCRIPTION
**Pull request by artem-konovalov-personal**
Closed [#266](https://github.com/nesterone/sf-eng-init-gl-spring-2016/issues/266)

### CODE REVIEW

File: 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js [hunk 1]
```diff
@@ -17,6 +17,10 @@ ancestry.forEach(function (person) {
   byName[person.name] = person;
 });
 
-// → Your code here.
+console.log(average(ancestry.filter(function (property) {
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js]: completely unreadable, make your code self-explaining

**Comment by artem-konovalov-personal** [comment]
I have changed the structure of source code for its self-explanation

File: 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
 /* global ancestry */
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js]: Use this one 
https://github.com/nesterone/sf-eng-init-gl-spring-2016/blob/master/01-syntax/04-high-order-functions/mother-child-age/ageDifference.js#L1

Hide all your work inside `ageDifference`

**Comment by artem-konovalov-personal** [comment]
I have made my code to be hidden into the `ageDifference` function

File: 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
 
-ancestry.forEach(function (person) {
-  byName[person.name] = person;
-});
+function ageDifference(incomingAncestryArray) {
+  var byName = {};
+  var personsWithMother;
+  var diffMotherAndKidAge;
+
+  incomingAncestryArray.forEach(function (person) {
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/mother-child-age/ageDifference.js]: I recommend to use `reduce` here

**Pull request closed without merging**